### ISSUE TITLE
Issue with backpropagation

### ISSUE THREAD

**Issue by bhqv** [OpenSourceSprint/ml#3]
## Description
Issue with backpropagation. When attempting to train the model using backpropagation, it seems that the loss function doesn't decrease.

## Difficulty
- Intermediate
- 30 Points (FOR EACH CORRECTION)

## Expected Behavior
I expect the loss to decrease over epochs as the model learns from the data.

## Actual Behavior
The loss remains stagnant, and the model doesn't converge even after multiple epochs.

**HINT: Bringing clarity to complexity**, some secret issues needs resolving
